File: src/lib/pjecalc/core/comum/validators/compared.ts
/**
 * PJe-Calc v2.15.1 — Compared (annotation TS-adaptada)
 * Porte de: br.jus.trt8.pjecalc.negocio.comum.validators.Compared
 */
export interface ComparedSpec {
  with: string;
  result: number; // -1, 0, 1 (compareTo result)
  not?: boolean;
  groups?: number[];
  condition?: string;
  message?: string;
}

export interface Compared_ {
  with: string;
  result: number;
  not: boolean;
  groups: number[];
  condition: string;
  message: string;
}

export function compared(spec: ComparedSpec): Compared_ {
  return {
    with: spec.with,
    result: spec.result,
    not: spec.not ?? false,
    groups: spec.groups ?? [],
    condition: spec.condition ?? '',
    message: spec.message ?? '{MSG0004}',
  };
}

File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/comum/validators/Compared.java
/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  org.hibernate.validator.ValidatorClass
 */
package br.jus.trt8.pjecalc.negocio.comum.validators;

import br.jus.trt8.pjecalc.negocio.comum.validators.ComparedValidator;
import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import org.hibernate.validator.ValidatorClass;

@Target(value={ElementType.FIELD})
@Retention(value=RetentionPolicy.RUNTIME)
@ValidatorClass(value=ComparedValidator.class)
@Documented
public @interface Compared {
    public byte[] groups() default {};

    public String condition() default "";

    public String with();

    public boolean not() default false;

    public int result();

    public String message() default "{MSG0004}";
}

